theano/tensor/tests/test_opt.py

```diff
@@ -3547,6 +3548,34 @@ def test2(self):
 
         self._verify_stack_trace(f)
 
+    def test_useless_alloc_with_shape_one(self):
+        alloc_lift = out2in(local_useless_alloc)
+        x = shared(self.rng.randn(2,))
+        y = shared(self.rng.randn())
+        alloc_x = tensor.alloc(x, 1, 3, 2)
+        alloc_y = tensor.alloc(y, 1, 1)
+
+        g = FunctionGraph([x, y], [alloc_x, alloc_y])
+        self.assertTrue(str(g) == ("[Alloc(<TensorType(float64, vector)>, "
+                                   "TensorConstant{1}, "
+                                   "TensorConstant{3}, "
+                                   "TensorConstant{2}), "
+                                   "Alloc(<TensorType(float64, scalar)>, "
+                                   "TensorConstant{1}, "
+                                   "TensorConstant{1})]"))
+
+        alloc_lift.optimize(g)
+        self.assertTrue(str(g) == "[DimShuffle{x,0,1}"
+                                  "(Alloc(<TensorType(float64, vector)>, "
+                                  "TensorConstant{3}, "
+                                  "TensorConstant{2})), "
+                                  "DimShuffle{x,x}"
+                                  "(<TensorType(float64, scalar)>)]")
+
+        # Check stacktrace was copied over correctly after opt was applied
+        self.assertTrue(hasattr(g.outputs[0].tag, 'trace'))
```

Comment:
@lamblin , ```g``` is not a function but a ```FunctionGraph```. So instead of ```self.assertTrue(hasattr(output.variable.tag, 'trace'))```, it should be ```self.assertTrue(hasattr(output.tag, 'trace'))```.
If I convert ```g``` to a function, then I don't know how to check ```str(g)```.
What's your opinion?